common/utils/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala

```diff
@@ -74,29 +69,68 @@ private object ConfigHelpers {
     v.map(stringConverter).mkString(",")
   }
 
-  def timeFromString(str: String, unit: TimeUnit): Long = JavaUtils.timeStringAs(str, unit)
+  def timeFromString(str: String, unit: TimeUnit, key: String): Long = {
+    try {
+      JavaUtils.timeStringAs(str, unit)
+    } catch {
```

Comment:
Just a question: Is this new `try-catch` inevitable because we cannot have this inside `JavaUtils.timeStringAs`?

Reply:
Some non-ConfigEntry callers are using this, so I handle errors here to avoid improper INVALID_CONF_VALUE categorization for these, e.g.

- org.apache.spark.sql.execution.streaming.FileStreamOptions#maxFileAgeMs
- org.apache.spark.sql.execution.streaming.FileStreamSink#retention